### PR TITLE
2020 update

### PR DESCRIPTION
**Pull request by joelhulen**
Please review the significant changes made to this MCW.

### CODE REVIEW

**Comment by joelhulen** [comment]
Closes #44

File: Hands-on lab/HOL step-by step - Cosmos DB real-time advanced analytics.md [hunk 1]
```diff
@@ -79,24 +92,22 @@ Woodgrove Bank, who provides payment processing services for commerce, is lookin
...
-Data scientists and engineers can create their own reports against Databricks tables, using Azure Databricks notebooks.
+The data flow for the solution begins with the payment transaction systems writing transactions to Azure Cosmos DB. Using the built-in change feed feature in Cosmos DB, the transactions can be read as a stream of incoming data within an Azure Synapse Analytics notebook, using Synapse Link. All transactional data is automatically stored in a fully isolated column store by enabling the Azure Cosmos DB Analytical Store. This store enables large-scale analytics against the operational data in Azure Cosmos DB, without impacting the transactional workloads or incurring resource unit (RU) costs. Woodgrove Bank's analysts query historical data contained within the analytical store and use it to join on reference data stored within the analytical store of other containers, as well as the data lake.
```
Comment (Rodrigossz) [inline Hands-on lab/HOL step-by step - Cosmos DB real-time advanced analytics.md:97]: 1) In this paragraph, sounds like that change feed is creating the data in columnar format. At least for someone that is not native in English as I am. 

2) Replace "Analytical Store" with "analytical store". 

3) I would also mention Azure Synapse Spark and Azure Synapse SQL Serverless.

4) You would do a great favor explaining that the transactions container will be created with analytical store enabled. Maybe also explain analytical store ttl. Can you do that for us?

5) Another favor: explain that the Cosmos DB database account will be created with Synapse Link integration enabled.

File: Hands-on lab/HOL step-by step - Cosmos DB real-time advanced analytics.md [hunk 2]
```diff
@@ -474,578 +485,1243 @@ In this exercise, you will use the data generator to send data to both Event Hub
...
+        accountAge,
+        isUserRegistered,
+        id
+    FROM OPENROWSET
+        (
+            'CosmosDB',
+            N'account=YOUR_ACCOUNT_NAME;database=Woodgrove;key=YOUR_ACCOUNT_KEY',
+            metadata
+        )
+    WITH (
```
Comment (Rodrigossz) [inline Hands-on lab/HOL step-by step - Cosmos DB real-time advanced analytics.md:1206]: This WITH session won't be necessary in the next couple of months. Let's go as is now and change later. ok?

File: Hands-on lab/HOL step-by step - Cosmos DB real-time advanced analytics.md [hunk 3]
```diff
@@ -79,24 +92,22 @@ Woodgrove Bank, who provides payment processing services for commerce, is lookin
...
...ice running on a highly available Azure Kubernetes Service cluster (AKS cluster). Woodgrove Bank uses the batch-scoring machine learning model within Azure Synapse Analytics notebooks for predicting fraud against the day's transactions, building aggregates showing statistics around fraudulent activity, and writing the results to an Azure Cosmos DB container. The batch-scoring model is also used within a Synapse notebook to reduce prediction latency by scoring the streaming data from the Azure Cosmos DB change feed, using Spark Structured Streaming. All transactions with "suspicious activity" output are stored in Azure Cosmos DB so it is globally available in regions closest to Woodgrove Bank's customers through their web applications. The analytical store feature is enabled on the container that contains predicted suspicious activity. Synapse SQL Serverless views are created against this and other analytical stores and can be accessed by business analysts using dashboards and reports in Power BI, which are embedded within the Synapse Analytics workspace. Data scientists and engineers can create their own reports against the Azure Cosmos DB Analytical Store, using Synapse notebooks.
```
Comment (Rodrigossz) [inline Hands-on lab/HOL step-by step - Cosmos DB real-time advanced analytics.md:101]: "Analytical Store" instead of "analytical store". No upper case...

**Review by Rodrigossz — COMMENTED** [review]
Hey! Great job! Excellent... Please check all my comments, some small adjustments.

File: Hands-on lab/HOL step-by step - Cosmos DB real-time advanced analytics.md (hunk 2 above)
Comment (joelhulen) [reply Hands-on lab/HOL step-by step - Cosmos DB real-time advanced analytics.md:1206]: Sounds good to me.

File: README.md [hunk 4]
```diff
@@ -14,7 +14,7 @@ October 2019
 
 ### Workshop
 
-In this workshop, you will learn to design a data pipeline solution that leverages Cosmos DB for both the scalable ingest of streaming data, and the globally distributed serving of both pre-scored data and machine learning models. The solution leverages the Cosmos DB change data feed in concert with the Azure Databricks Delta to enable a modern data warehouse solution that can be used to create risk reduction solutions for scoring transactions for fraud in an offline, batch approach and in a near real-time, request/response approach.
+In this workshop, you will learn to design a data pipeline solution that leverages Cosmos DB for both the scalable ingest of streaming data, and the globally distributed serving of both pre-scored data and machine learning models. The solution leverages the Cosmos DB change data feed in concert with the Azure Synapse Analytics to enable a modern data warehouse solution that can be used to create risk reduction solutions for scoring transactions for fraud in an offline, batch approach and in a near real-time, request/response approach.
```
[the remark this answers is not in the document]
Comment (joelhulen) [reply README.md:17]: Making this change now. Thanks for the suggestion.

File: Hands-on lab/Deployment/environment-template-cosmos.json [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+    "parameters": {
+        "uniqueSuffix": {
+            "type": "String",
+            "metadata": {
+                "description": "Suffix added to all resource names to make them unique."
+            }
+        },
+        "throughputPolicy": {
+            "type": "string",
+            "defaultValue": "Autoscale",
```
Comment (markjbrown) [inline Hands-on lab/Deployment/environment-template-cosmos.json:13]: That's a mighty nice ARM template :)

File: Hands-on lab/HOL step-by step - Cosmos DB real-time advanced analytics.md (hunk 1 above)
Comment (markjbrown) [reply Hands-on lab/HOL step-by step - Cosmos DB real-time advanced analytics.md:97]: I agree with Rodrigo here. The data flow described seems to indicate that change feed is writing the data.

File: Hands-on lab/Deployment/environment-template-cosmos.json (hunk 5 above)
Comment (joelhulen) [reply Hands-on lab/Deployment/environment-template-cosmos.json:13]: Thanks, @markjbrown :) I was inspired by your awesome Cosmos DB templates.

**Comment by joelhulen** [comment]
@Rodrigossz and @markjbrown - Thanks so much for your feedback! I've committed changes to my fork, and by extension, this PR, based on your comments. Please review my changes at your earliest convenience. Thanks!

File: Hands-on lab/HOL step-by step - Cosmos DB real-time advanced analytics.md [hunk 6]
```diff
@@ -79,24 +92,24 @@ Woodgrove Bank, who provides payment processing services for commerce, is lookin
...
...time scoring web service running on a highly available Azure Kubernetes Service cluster (AKS cluster). Woodgrove Bank uses the batch-scoring machine learning model within Azure Synapse Analytics notebooks to predict fraud against the day's transactions, build aggregates showing statistics around fraudulent activity, and write the results to an Azure Cosmos DB container. The batch-scoring model is also used within a Synapse notebook to reduce prediction latency by scoring the Azure Cosmos DB change feed's streaming data, using Spark Structured Streaming. All transactions with "suspicious activity" output are stored in Azure Cosmos DB, so it is globally available in regions closest to Woodgrove Bank's customers through their web applications. The analytical store feature is enabled on the container that contains predicted suspicious activity. Synapse SQL Serverless views are created against this and other analytical stores. Business analysts can access them using dashboards and reports in Power BI, which are embedded within the Synapse Analytics workspace. Data scientists and engineers can create their own reports against the Azure Cosmos DB analytical store, using Synapse notebooks.
```
Comment (Rodrigossz) [inline Hands-on lab/HOL step-by step - Cosmos DB real-time advanced analytics.md:103]: Hello! Good morning! Synapse names JUST CHANGED. New naming arrived yesterday and you may want to publish with the new rules...

NEW FULL NAMES
Azure Synapse serverless Apache Spark pools
Azure Synapse serverless SQL pools
Azure Synapse dedicated SQL pool

Both "serverless" and "dedicated" are all lower case.

When you are mentioning the capability, not the product, you can use:
serverless Apache Spark pools
serverless SQL pools
dedicated SQL pools

OK?

**Review by Rodrigossz — COMMENTED** [review]
Please check the new names.

Hello! Good morning! Synapse names JUST CHANGED. New naming arrived yesterday and you may want to publish with the new rules...

NEW FULL NAMES
Azure Synapse serverless Apache Spark pools
Azure Synapse serverless SQL pools
Azure Synapse dedicated SQL pool

Both "serverless" and "dedicated" are all lower case.

When you are mentioning the capability, not the product, you can use:
serverless Apache Spark pools
serverless SQL pools
dedicated SQL pools

OK?

File: Hands-on lab/HOL step-by step - Cosmos DB real-time advanced analytics.md (hunk 6 above)
Comment (joelhulen) [reply Hands-on lab/HOL step-by step - Cosmos DB real-time advanced analytics.md]: Oh wow, great timing! I've updated the lab with the new names. Can you please verify?

**Review by Rodrigossz — APPROVED** [review]
Great job!

**Pull request merged**